runtime/lua/vim/diagnostic.lua

```diff
@@ -101,6 +101,7 @@ end
 --- @field update_in_insert boolean
 --- @field underline vim.diagnostic.Opts.Underline
 --- @field virtual_text vim.diagnostic.Opts.VirtualText
+--- @field virtual_lines vim.diagnostic.Opts.VirtualLines
```

Comment:
We also need to add a `virtual_lines` entry to `vim.diagnostic.Opts` (after line 74)